pkgs/build-support/fetchgit/nix-prefetch-git

```diff
@@ -150,10 +152,16 @@ url_to_name(){
     local base
     base=$(basename "$url" .git | cut -d: -f2)
 
-    if [[ $ref =~ ^[a-z0-9]+$ ]]; then
-        echo "$base-${ref:0:7}"
+    if ! test -z "$rootDir"; then
```

Comment:
No need for the double-negative here (`!` and `-z`): this should be `test "$rootDir"` or `test -n "$rootDir"`.

(Personally I'd prefer to use `[[ ... ]]` tests, but the standard in this script is clearly Bash's `test` builtin.)